web/src/components/pages/profile/info/info.tsx

```diff
@@ -193,6 +184,20 @@ function ProfilePage({
     return null;
   }
 
+  if (!isSaving && isSubmitted && isEnrolled(user.account)) {
+    return (
+      <Redirect
+        to={{
+          pathname: toLocaleRoute(URLS.DASHBOARD + URLS.CHALLENGE),
+          state: {
+            showOnboardingModal: true,
+            earlyEnroll: window.location.search.includes('achievement=1'),
+          },
```

Comment:
I like this so much better than just endlessly passing url params around, yay!

Reply:
👍